spring-batch-integration/src/test/java/org/springframework/batch/integration/partition/RemotePartitioningMasterStepBuilderTest.java

```diff
@@ -0,0 +1,273 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.core.MessagingTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.springframework.test.util.ReflectionTestUtils.getField;
+
+/**
+ * @author Mahmoud Ben Hassine
+ */
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {RemotePartitioningMasterStepBuilderTest.BatchConfiguration.class})
+public class RemotePartitioningMasterStepBuilderTest {
+
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+
+	@Autowired
+	private JobRepository jobRepository;
+
+	@Test
+	public void inputChannelMustNotBeNull() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("inputChannel must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.inputChannel(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void outputChannelMustNotBeNull() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("outputChannel must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.outputChannel(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void messagingTemplateMustNotBeNull() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("messagingTemplate must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.messagingTemplate(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void jobExplorerMustNotBeNull() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("jobExplorer must not be null");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.jobExplorer(null)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void pollIntervalMustBeGreaterThanZero() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("The poll interval must be greater than zero");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.pollInterval(-1)
+				.build();
+
+		// then
+		// expected exception
+	}
+
+	@Test
+	public void timeoutMustNotBeGreaterThanZero() {
+		// given
+		this.expectedException.expect(IllegalArgumentException.class);
+		this.expectedException.expectMessage("The timeout must be greater than zero");
+
+		// when
+		Step step = new RemotePartitioningMasterStepBuilder("step")
+				.timeout(-1L)
```

Comment:
Is there the option for no timeout?

Reply:
Changed the default to `-1L` which means no timeout (like in `org.springframework.batch.integration.partition.MessageChannelPartitionHandler#setTimeout`)